### PR TITLE
GH-37934: [Doc][Integration] Document C Data Interface testing

### LINKED ISSUE #37934: [Doc][Integration] Document C Data Interface testing
### Describe the enhancement requested

Following gh-37537, we want to update the integration testing documentation to also cover the C Data Interface.

### Component(s)

Documentation, Integration

### PR DESCRIPTION
**Pull request by pitrou**
### Rationale for this change

gh-37537 added integration testing for the C Data Interface, but the documentation was not updated.

### What changes are included in this PR?

Add documentation for C Data Interface integration testing.

### Are these changes tested?

N/A, only doc changes.

### Are there any user-facing changes?

No.
* Closes: #37934

### CODE REVIEW

**Comment by pitrou** [comment]
@github-actions crossbow submit preview-docs

File: docs/source/format/Integration.rst [hunk 1]
```diff
@@ -20,32 +20,97 @@
...
+* Each (producer, consumer) pair is tested over a range of JSON files
+  representing different data type categories, such as numerics, lists, etc.
+  This makes it easier to pinpoint incompatibilities than if all data types
+  were represented in a single file.
+
+Example: IPC format
+~~~~~~~~~~~~~~~~~~~
+
+Let's say we are testing Arrow C++ as a producer and Arrow Java as a consumer
+of the Arrow IPC format. Testing a JSON file would go as follows:
```
Comment (bkietz) [inline docs/source/format/Integration.rst:60]: Should we mention `datagen.py` here or in the strategy section? Without the mention a reader might expect the JSON files to be checked in somewhere

File: docs/source/format/Integration.rst [hunk 2]
```diff
@@ -20,32 +20,97 @@
...
+
+#. A C++ executable reads the JSON file, converts it into Arrow in-memory data
+   and writes a Arrow IPC file (the file paths are typically given on the command
```
Comment (bkietz) [inline docs/source/format/Integration.rst:63]:
```suggestion
   and writes aa Arrow IPC file (the file paths are typically given on the command
```

File: docs/source/format/Integration.rst [hunk 3]
```diff
@@ -20,32 +20,97 @@
...
+   line).
+
+#. A Java executable reads the JSON file, converts it into Arrow in-memory data;
+   it also reads the Arrow IPC file. Finally, it validates that both Arrow
```
Comment (bkietz) [inline docs/source/format/Integration.rst:67]:
```suggestion
   it also reads the Arrow IPC file generated by C++. Finally, it validates that both Arrow
```

**Review by bkietz — COMMENTED** [review]
Looks solid, just a few nits

File: docs/source/format/Integration.rst (hunk 2 above)
Comment (pitrou) [reply docs/source/format/Integration.rst:63]: Hmm, is your suggestion a typo? Did you mean "an"?

File: docs/source/format/Integration.rst (hunk 1 above)
Comment (pitrou) [reply docs/source/format/Integration.rst:60]: I'll mention data generation, though a couple of them are checked in as well (the "gold files").

File: docs/source/format/Integration.rst (hunk 2 above)
Comment (bkietz) [reply docs/source/format/Integration.rst:63]: yep, oops

**Review by jorisvandenbossche — APPROVED** [review]
Not being super familiar with this, but I find everything clearly explained to me

**Review by bkietz — APPROVED** [review]

**Pull request merged**